app/src/main/java/org/eyeseetea/malariacare/data/database/model/Survey.java

```diff
@@ -551,30 +549,39 @@ private long countNumOptionalQuestionsToAnswer() {
     /**
      * Updates ratios, status and completion date depending on the question and answer (text)
      */
-    public void updateSurveyStatus() {
+    public void updateSurveyStatus(GetSurveyAnsweredRatioUseCase.RecoveryFrom recoveryFrom) {
 
         //Exit if the survey was sent or completed
         if (isReadOnly()) {
             return;
         }
 
-
-        SurveyAnsweredRatio answeredRatio = this.reloadSurveyAnsweredRatio(null);
-
-        SurveyAnsweredRatio surveyAnsweredRatio = this.getAnsweredQuestionRatio();
-        if (surveyAnsweredRatio.getTotalCompulsory() == 0) {
-            //Update status
-            if (!answeredRatio.isCompleted()) {
-                this.setStatus(Constants.SURVEY_IN_PROGRESS);
-            }
-
-        } else if (surveyAnsweredRatio.getCompulsoryAnswered() == 0) {
-            this.setStatus(Constants.SURVEY_IN_PROGRESS);
-        }
-
-
-        //Saves new status & completion_date
-        this.save();
+        GetSurveyAnsweredRatioUseCase getSurveyAnsweredRatioUseCase = new GetSurveyAnsweredRatioUseCase();
```

Comment:
@idelcano an UseCase can not be invoked from data layer, Use case should be invoked only from presentation layer. On this way updateSurveyStatus has more of one resposability. A possible option is from where we invoke updateSurveyStatus first we calculate SurveyAnsweredRatio and depending on the result invoke updateSurveyStatus with the desired state